tests/test_ls.py

```diff
@@ -147,12 +145,57 @@ def test_ls_dircache_remove_cached(
     prefix = f"{repository}/{temp_branch}"
     resource = f"{prefix}/"
 
-    try:
-        listing_pre = fs.ls(resource)
-        fs.rm(listing_pre[0]["name"])
+    listing_pre = fs.ls(resource)
+    fs.rm(listing_pre[0]["name"])
 
-        # List again, cache should have been invalidated by rm
-        listing_post = fs.ls(resource)
-        assert len(listing_post) == len(listing_pre) - 1
-    finally:
-        client_helpers.reset_branch(fs.client, repository, temp_branch)
+    # List again, cache should have been invalidated by rm
+    listing_post = fs.ls(resource)
+    assert len(listing_post) == len(listing_pre) - 1
+
+
+def test_ls_dircache_recursive(
+    fs: LakeFSFileSystem,
+    repository: str,
+    temp_branch: str,
+) -> None:
+    prefix = f"{repository}/{temp_branch}"
+
+    # (1) Basic recursive dircache filling
+    fs.dircache.clear()
```

Comment:
Should we move this to the `fs` fixture?

Reply:
It's not even required, since we clear the instance cache and return a fresh FS instance in the fixture.

Removed in 926fb3be2d1eee23b75dfc9514e8570acff35519.